kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/dek/Aes.java

```diff
@@ -20,22 +20,29 @@
 
 public class Aes implements CipherManager {
 
-    public static final Aes AES_256_GCM_128 = new Aes("AES_256/GCM/NoPadding", (byte) 0, CipherSpec.AES_256_GCM_128);
+    public static final Aes AES_256_GCM_128 = new Aes("AES/GCM/NoPadding", 256, (byte) 0, CipherSpec.AES_256_GCM_128);
```

Comment:
I think this is technically a breaking change as we consider the config model part of the public API and the old name is part of that API. Do we need alias mechanism?

Reply:
I'm confused, where is the transformation String used in config?

Reply:
OH is it not? I thought we allowed people to specify it. If not then ignore me 😁

Reply:
I _think_ it's currently hardcoded to AES somewhere, maybe this blob in the EncryptionDekCache.

```
private static <K> CacheKey<K> cacheKey(EncryptionScheme<K> encryptionScheme) {
        return new CacheKey<>(encryptionScheme.kekId(), CipherSpec.AES_256_GCM_128);
    }
```

I guess when we make it configurable it would be by cipherspec name rather than the underlying transform.

Reply:
Rob's correct. The `CipherSpec` is currently not configurable. And even if it was that's not the same thing as a configurable transformation for the call to `Cipher .getInstance()`.

Reply:
Sweet! always glad to be wrong.